File: indexer-desktop-applicaiton/src/api-calls/hits/hits.ts
import { HitsRequest, HitsResponse } from "./hits.type";
import { getBaseUrlFromElectron } from "../utils";

export async function getHits(request: HitsRequest): Promise<HitsResponse> {
    const baseUrl = await getBaseUrlFromElectron();
    const response = await fetch(`${baseUrl}/hits`, {
        method: 'POST',
        headers: {
            'Content-Type': 'application/json',
            'Authorization': `Bearer ${request.accessToken}`,
        },
        body: JSON.stringify({
            tc_query_id: request.tc_query_id,
            duration: request.duration,
            optimized: request.optimized,
        }),
    });

    if (!response.ok) {
        throw new Error('Failed to retrieve hits');
    }

    const data = await response.json();
    return data as HitsResponse;
}

File: indexer-api/app/schemas/hits.py
from pydantic import BaseModel
from typing import List

class HitsRequest(BaseModel):
    tc_query_id: int
    duration: int
    optimized: bool = True

class QueryLog(BaseModel):
    id: int
    time_stamp: str
    optimized: bool

class HitsResponse(BaseModel):
    query_logs: List[QueryLog]
